### ISSUE TITLE
Establish and document deprecation protocol

### ISSUE THREAD

**Issue by raoulcollenteur** [pastas/pastas#546]
With Pastas 1.0, we should establish a more formal protocol for deprecations and removal of old functions. 

I think we should copy this from one of the packages in the Scipy stack, but still put it in our documentation under "developers" and familiarise everyone with the process. 

- [x] Develop protocol (Here in this issue).
- [x] Document in rst file.
- [x] Make code that we can use to do this to make the process similar each time.

Ideas and help on this more than welcome.

**Comment by martinvonk** [comment]
How to raise FutureWarning in logger:

```python
from logging import captureWarnings, getLogger
from warnings import warn

captureWarnings(True)
logger = getLogger(__name__)
warnings_logger = getLogger("py.warnings")
warnings_logger.addHandler(logger)

warn(
    "Type Your Message Here",
    category=FutureWarning,
)
```

**Comment by dbrakenhoff** [comment]
Closed by #731
